charts/fluentd/templates/service.yaml

```diff
@@ -1,4 +1,4 @@
-
+{{- if and (eq .Values.kind "DaemonSet") .Values.service.enabled -}}
```

Comment:
What is the reason to compare '.Values.kind "DaemonSet' and  '.Values.service.enabled'.

Shouldn't the latter one be enough?

Reply:
I didn't want to disturb the use-cases as StatefulSet or Deployment because I don't use them or know about their configuration, so I put the additional check to limit the scope of this change to only the DaemonSet.

Reply:
well, this is not a viable solution as it will behave differently depending on the controller you use.

Remove it from only DaemonSet and ideally test the other options if possible...

Reply:
I see the concern you are raising now. It's because it will not deploy the service in the StatefulSet and Deployment controllers. I can address this so it will be the same as before in the case of StatefulSet and Deployment (always deploy the service) but give the option to configure true/false to deploy the service with the DaemonSet if this will be acceptable.

Reply:
Submit the PR with the new option, but without discriminating controllers. Should be fine .TM